### PR TITLE
adding notebook for bootstrap

### PR DESCRIPTION
**Pull request by sumedha-k**
The code is in testNotebook2.pynb @aarmey

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
You're resetting `random_state=0` on every iteration, so all your resamplings are identical.

**Comment by sumedha-k** [comment]
Oh my goodness- I completely forgot to take that out

**Comment by sumedha-k** [comment]
@aarmey here's the new graph, I currently have the sample size set to 40, should I change this number at all?

**Comment by sumedha-k** [comment]
If everything looks good, should I add it to a new python file under figures?

**Comment by aarmey** [comment]
@sumedha-k I think you can add this to figure1.py!

**Comment by sumedha-k** [comment]
@aarmey The build error is "no module named 'sklearn'"

**Comment by aarmey** [comment]
Add `scikit-learn==0.24.1` to `requirements.txt`

**Comment by sumedha-k** [comment]
@aarmey I attempted to commute the error terms using the ols solver that we discussed last week. I know you suggested that I use np.linalg.pinv(X) for (X' X)^{-1}, but I don't think they are functionally equivalent because pinv returns a non-square matrix. Regardless, I think I have a bug because my error terms are all really small, so I was hoping you could take a look. Everything is in testNotebook2.pynb starting from cell 7.

File: deconv/figures/figure1.py [hunk 1]
```diff
@@ -30,3 +32,47 @@ def makeFigure():
     subplotLabel(ax)
 
     return f
+
+def makeFigureWithError():
```
Comment (aarmey) [inline deconv/figures/figure1.py:36]: Make a `figure2.py` file.

File: deconv/figures/figure1.py [hunk 2]
```diff
@@ -30,3 +32,47 @@ def makeFigure():
...
+        mean_3a = new3a.groupby(level=0).sum() / new3a.groupby(level=0).count()
+
+        new3b = resample(adcc_3b, n_samples=40, replace=True, stratify=adcc_3b.index)
+        mean_3b = new3b.groupby(level=0).sum() / new3b.groupby(level=0).count()
+
+        glycans_3a.append(infer_x(A_antiD, mean_3a))
+        glycans_3b.append(infer_x(A_antiD, mean_3b))
+    
+    glycans3a = np.array(glycans_3a)
+    mean3a = np.mean(glycans3a, axis=0)
```
Comment (aarmey) [inline deconv/figures/figure1.py:58]: The center of a bootstrap plot should not be the mean of samples, it should be the result without resampling.

File: deconv/figures/figure1.py [hunk 3]
```diff
@@ -30,3 +32,47 @@ def makeFigure():
...
+    error3a = np.std(glycans3a, axis=0)
```
Comment (aarmey) [inline deconv/figures/figure1.py:59]: The error bars needn't be symmetric. Take the 33rd and 66th quantiles to get lower and upper bar ranges. If `yerr` gives two values then it plots asymmetric error bars.

**Review by aarmey — COMMENTED** [review]
I think the issue is that the newer way only quantifies the data uncertainty, and virtually all of the uncertainty comes from the variance of the model. We can use bootstrapping—I've pointed out some changes to make, then we can get this merged.

**Comment by aarmey** [comment]
These were quick fixes so I took care of them. Merging!

**Pull request merged**